### PR TITLE
feat: add docs for Bitbucket Server external auth config

### PR DESCRIPTION
**Pull request by ericpaulsen**
this PR adds documentation for connecting Bitbucket Server as an external auth provider with Coder. it also includes clarifications around the GitHub Enterprise config.

### CODE REVIEW

**Review by Emyrk — APPROVED** [review]

**Comment by matifali** [comment]
@ericpaulsen I suggest we also document all supported values for `CODER_EXTERNAL_AUTH_0_TYPE`.

**Review by matifali — APPROVED** [review]

File: docs/admin/external-auth.md [hunk 1]
```diff
@@ -70,14 +70,30 @@ CODER_EXTERNAL_AUTH_0_DISPLAY_ICON="https://mycustomicon.com/google.svg"
...
+### Bitbucket Server
+
+Bitbucket Server requires the following environment variables:
+
+```env
+CODER_EXTERNAL_AUTH_0_TYPE="bitbucket-server"
+CODER_EXTERNAL_AUTH_0_ID=bitbucket
+CODER_EXTERNAL_AUTH_0_CLIENT_ID=xxx
+CODER_EXTERNAL_AUTH_0_CLIENT_SECRET=xxx
+CODER_EXTERNAL_AUTH_0_AUTH_URL=https://bitbucket.domain.com/rest/oauth2/latest/authorize
```
Comment (matifali) [inline docs/admin/external-auth.md:94]: Doesn't bitbucket-server need the `CODER_EXTERNAL_AUTH_0_TOKEN_URL`?

**Comment by Emyrk** [comment]
> @ericpaulsen I suggest we also document all supported values for `CODER_EXTERNAL_AUTH_0_TYPE`.

I see the types here: https://coder.com/docs/v2/latest/admin/external-auth#configuration

File: docs/admin/external-auth.md (hunk 1 above)
Comment (Emyrk) [reply docs/admin/external-auth.md:94]: PR that does this https://github.com/coder/coder/pull/10520

**Pull request merged**